### PR TITLE
Generate and check for distinct maps in scrimmage requestor

### PR DESCRIPTION
**Pull request by gaurav-arya**

### CODE REVIEW

File: frontend/src/components/scrimmageRequestForm.js [hunk 1]
```diff
@@ -31,12 +31,22 @@ class ScrimmageRequestForm extends Component {
     this.requestScrimmage = this.requestScrimmage.bind(this);
   }
 
+  getRandomMaps(available_maps) {
+    const possible_maps = available_maps.slice();
```
Comment (n8kim1) [inline frontend/src/components/scrimmageRequestForm.js:35]: you're sure this deep-copies right? potential way to test would be to run this multiple times

**Review by n8kim1 — APPROVED** [review]
looks mostly fine, see comment but ya

**Comment by n8kim1** [comment]
also consider regenerating random maps after each scrim req, for variety
(so that users don't have to refresh pg or smth)

File: frontend/src/components/scrimmageRequestForm.js (hunk 1 above)
Comment (gaurav-arya) [reply frontend/src/components/scrimmageRequestForm.js:35]: We don't need a deep copy since we aren't modifying the inner structs, just the array itself

**Comment by gaurav-arya** [comment]
We do regenerate the maps every time the modal is closed!

**Pull request merged**